homework/index.js

```diff
@@ -30,13 +30,122 @@
     return elem;
   }
 
+  function makeHeader(repos) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    header.style = 'display: flex;';
+    createAndAppend('h3', header, { text: 'HYF Repositories' });
+    const selectMenu = createAndAppend('select', header, { class: 'repo-selector' });
+    const optionsArr = [];
+    for (let i = 0; i < repos.length; i++) {
+      const option = {
+        name: repos[i].name,
+        index: i,
+      };
+      optionsArr.push(option);
+    }
+    optionsArr
+      .sort((a, b) => a.name.localeCompare(b.name, { caseFirst: 'lower' }))
+      .map(element =>
+        createAndAppend('option', selectMenu, {
+          value: element.index,
+          text: element.name,
+          id: element.name,
+        }),
+      );
+    return header;
+  }
+
+  function makeTable(repo) {
+    const leftDiv = document.querySelector('.leftDiv');
+    const table = createAndAppend('table', leftDiv);
+    const tBody = createAndAppend('tbody', table);
+    const trRepo = createAndAppend('tr', tBody);
+    createAndAppend('td', trRepo, { text: 'repo:' });
+    const tdRepoLink = createAndAppend('td', trRepo);
+    createAndAppend('a', tdRepoLink, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    if (repo.description !== null) {
+      const trDescription = createAndAppend('tr', tBody, {});
+      createAndAppend('td', trDescription, { text: 'Description:' });
+      createAndAppend('td', trDescription, { text: repo.description });
+    }
+    const trForks = createAndAppend('tr', tBody);
+    createAndAppend('td', trForks, { text: 'Forks:' });
+    createAndAppend('td', trForks, { text: repo.forks_count });
+    const trUpdate = createAndAppend('tr', tBody);
+    createAndAppend('td', trUpdate, { text: 'Updated:' });
+    createAndAppend('td', trUpdate, {
+      text: new Date(repo.updated_at).toLocaleString(),
+    });
+  }
+
+  function makeUl(repos) {
```

Comment:
Try to think of more descriptive names. What will this `ul` contain? Apples? Or perhaps Contributors? In the latter case `makeContributorList` would more clearly describe what this function does.

Reply:
ok.